### PR TITLE
chore: projen subprojects

### PR DESCRIPTION
**Pull request by eladb**
Convert packages to use projen subprojects. This means that now we have a single projen entry point in the repo (the root `.projenrc.js`) which references `.subprojenrc.js` files within each module.

This is the first step in moving the project to projen completely. The remaining work is to codify the various repo-level configuration (workflows, etc) as part of the root project.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,108 +1,3 @@
-dist/
-.gen
-.vscode/
```
Comment (iliapolo) [inline .gitignore:3]: Lets keep this

File: .gitignore [hunk 2]
```diff
@@ -1,108 +1,3 @@
...
-pids
-*.pid
-*.seed
-*.pid.lock
-
-# Directory for instrumented libs generated by jscoverage/JSCover
-lib-cov
-
-# Coverage directory used by tools like istanbul
-coverage
```
Comment (iliapolo) [inline .gitignore:36]: And this

File: packages/cdk8s-plus-17/.subprojenrc.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      distName: `cdk8s-plus-${SPEC_VERSION}`,
+      module: `cdk8s_plus_${SPEC_VERSION}`
+    },
+    dotnet: {
+      dotNetNamespace: `Org.Cdk8s.Plus${SPEC_VERSION}`,
+      packageId: `Org.Cdk8s.Plus${SPEC_VERSION}`
+    }
+  });
+ 
+  project.addScript('import', `../cdk8s-cli/bin/cdk8s -l typescript -o ${IMPORTDIR} import k8s@${K8S_VERSION}`);
```
Comment (iliapolo) [inline packages/cdk8s-plus-17/.subprojenrc.js:46]: Nice

File: packages/cdk8s-cli/src/cli/cmds/import.ts [hunk 4]
```diff
@@ -28,9 +28,8 @@ class Command implements yargs.CommandModule {
 
   public async handler(argv: any) {
     const classNamePrefix = argv.classPrefix === false ? '' : argv.classPrefix;
-    const imports: string[] = Array.isArray(argv.spec) ? argv.spec : [argv.spec];
```
Comment (iliapolo) [inline packages/cdk8s-cli/src/cli/cmds/import.ts:31]: Where did this come from?

**Comment by iliapolo** [comment]
@eladb Also this is now broken:

https://github.com/awslabs/cdk8s/blob/a191c596a20cfaab030f79f9e46989c512fa2897/package.json#L9

Change to `"projen": "projen"`?

**Comment by eladb** [comment]
superseded by #436

**Pull request closed without merging**